src/components/Api.js

```diff
@@ -0,0 +1,99 @@
+/* Project 9 Overview:
+
+1. Acquire personal token
+2. Create new branch
+    - "git checkout -b project-9"
+3. Pass the token through "Around the US" project server (on project page)
+4. Create Api class
+    - All requests should be methods in this class
+    - User routes:
+	GET /users/me (get current user's info)
+	PATCH /users/me (update profile info)
+	PATCH /users/me/avatar (update avatar)
+    - Card routes:
+	GET /cards (get all cards)
+	POST /cards (create a card)
+	DELETE /cards/:cardId (delete a card)
+	PUT /cards/:cardId/likes (like a card)
+	DELETE /cards/:cardId/likes (dislike a card)
+5. See project page for details
+*/
+
+export default class Api {
+  constructor(options) {
+    this.baseUrl = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this.baseUrl}/cards`, {
+      headers: this.headers,
+    }).then(this._checkResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.